lib/puppet/type/netapp_qtree.rb

```diff
@@ -8,23 +8,54 @@
   ensurable
 
   newparam(:name) do
-    desc "The qtree name."
     isnamevar
+    desc "The qtree name."
     validate do |value|
-      unless value =~ /^[\w\-]+$/
+      unless (value =~ /^[\w\-\.]+$/) or (value =~ /^\/\w+\/[\w\-\.]+$/)
          raise ArgumentError, "%s is not a valid qtree name." % value
       end
     end
   end
 
   newparam(:volume) do
     desc "The volume to create qtree against."
+    defaultto do
+      if @resource[:name].count('/') == 2
+        @resource[:name].split('/')[1] 
+      else
+        ""
+      end
+    end
```

Comment:
@fduranti Sorry, busy day. It looks like the unit tests (run with `rake spec`) are a bit stale and are failing, but I made https://github.com/fduranti/puppetlabs-netapp/pull/1

I got bogged down getting the vsim running for development (I use https://github.com/tlichten/vagrant-vsim but haven't done it for a while) so haven't tested against an actual device yet.

I think you're right, what you're probably seeing is the `self.instances` values returned by the provider, though once that's matching correctly then `exists?` should also work as-is. I took a stab at updating `self.instances`, but can only hope it's accurate until I get a vsim going.